packages/aws-library/src/aws_library/s3/_client.py

```diff
@@ -0,0 +1,460 @@
+import asyncio
+import contextlib
+import logging
+import urllib.parse
+from collections.abc import AsyncGenerator, Callable, Sequence
+from dataclasses import dataclass, field
+from pathlib import Path
+from typing import Any, Final, cast
+
+import aioboto3
+from aiobotocore.session import ClientCreatorContext
+from boto3.s3.transfer import TransferConfig
+from botocore import exceptions as botocore_exc
+from botocore.client import Config
+from models_library.api_schemas_storage import ETag, S3BucketName, UploadedPart
+from models_library.basic_types import SHA256Str
+from pydantic import AnyUrl, ByteSize, parse_obj_as
+from servicelib.logging_utils import log_catch, log_context
+from servicelib.utils import limited_gather
+from settings_library.s3 import S3Settings
+from types_aiobotocore_s3 import S3Client
+from types_aiobotocore_s3.literals import BucketLocationConstraintType
+from types_aiobotocore_s3.type_defs import ObjectIdentifierTypeDef
+
+from ._constants import MULTIPART_UPLOADS_MIN_TOTAL_SIZE
+from ._error_handler import s3_exception_handler, s3_exception_handler_async_gen
+from ._errors import S3DestinationNotEmptyError, S3KeyNotFoundError
+from ._models import (
+    MultiPartUploadLinks,
+    S3DirectoryMetaData,
+    S3MetaData,
+    S3ObjectKey,
+    UploadID,
+)
+from ._utils import compute_num_file_chunks
+
+_logger = logging.getLogger(__name__)
+
+_S3_MAX_CONCURRENCY_DEFAULT: Final[int] = 10
+_DEFAULT_AWS_REGION: Final[str] = "us-east-1"
+_MAX_ITEMS_PER_PAGE: Final[int] = 500
+_MAX_CONCURRENT_COPY: Final[int] = 4
+_AWS_MAX_ITEMS_PER_PAGE: Final[int] = 1000
+
+
+@dataclass(frozen=True)
+class SimcoreS3API:  # pylint: disable=too-many-public-methods
```

Comment:
MIONR: I would go for `S3Api` or `AwsS3Api` (or analogous) but no need for `Simcore` prefix